junit-jupiter-params/src/test/java/org/junit/jupiter/params/ParameterizedTestIntegrationTests.java

```diff
@@ -1070,6 +1072,101 @@ private EngineExecutionResults execute(String methodName, Class<?>... methodPara
+			var results = execute("testWithRepeatableCsvSource", String.class);
+			results.allEvents().assertThatEvents() //
+					.haveExactly(1, event(test(), displayName("[1] argument=a"), finishedWithFailure(message("a")))) //
+					.haveExactly(1, event(test(), displayName("[2] argument=b"), finishedWithFailure(message("b"))));
+		}
+
+		@Test
+		void executesWithRepeatableMethodSource() {
+			var results = execute("testWithRepeatableMethodSource", String.class);
+			results.allEvents().assertThatEvents() //
+					.haveExactly(1,
+						event(test(), displayName("[1] argument=some"), finishedWithFailure(message("some")))) //
+					.haveExactly(1,
+						event(test(), displayName("[2] argument=other"), finishedWithFailure(message("other"))));
+		}
+
+		@Test
+		void executesWithRepeatableEnumSource() {
+			var results = execute("testWithRepeatableEnumSource", Action.class);
+			results.allEvents().assertThatEvents() //
+					.haveExactly(1, event(test(), displayName("[1] argument=FOO"), finishedWithFailure(message("FOO")))) //
+					.haveExactly(1,
+						event(test(), displayName("[2] argument=BAR"), finishedWithFailure(message("BAR"))));
+		}
+
+		@Test
+		void executesWithRepeatableValueSource() {
+			var results = execute("testWithRepeatableValueSource", String.class);
+			results.allEvents().assertThatEvents() //
+					.haveExactly(1, event(test(), displayName("[1] argument=foo"), finishedWithFailure(message("foo")))) //
+					.haveExactly(1,
+						event(test(), displayName("[2] argument=bar"), finishedWithFailure(message("bar"))));
+		}
+
+		@Test
+		void executesWithRepeatableFieldSource() {
+			var results = execute("testWithRepeatableFieldSource", String.class);
+			results.allEvents().assertThatEvents() //
+					.haveExactly(1,
+						event(test(), displayName("[1] argument=some"), finishedWithFailure(message("some")))) //
+					.haveExactly(1,
+						event(test(), displayName("[2] argument=other"), finishedWithFailure(message("other"))));
+		}
+
+		@Test
+		void executesWithRepeatableArgumentsSource() {
+			var results = execute("testWithRepeatableArgumentsSource", String.class);
+			results.allEvents().assertThatEvents() //
+					.haveExactly(1, event(test(), displayName("[1] argument=foo"), finishedWithFailure(message("foo")))) //
+					.haveExactly(1, event(test(), displayName("[2] argument=bar"), finishedWithFailure(message("bar")))) //
+					.haveExactly(1, event(test(), displayName("[3] argument=foo"), finishedWithFailure(message("foo")))) //
+					.haveExactly(1,
+						event(test(), displayName("[4] argument=bar"), finishedWithFailure(message("bar"))));
+
+		}
+
+		@Test
+		void executesWithSameRepeatableAnnotationMultipleTimes() {
+			var results = execute("testWithSameRepeatableAnnotationMultipleTimes", String.class);
+			results.allEvents().assertThatEvents() //
+					.haveExactly(1, event(test(), started())) //
+					.haveExactly(1, event(test(), finishedWithFailure(message("foo"))));
```

Comment:
I think we should also check for the "repeated" invocations here.

Reply:
@madalingiurca Did you see this comment?

Reply:
The name of the test might be misleading, because we don't expect repeated invocations here.
The purpose of this test is to validate that when the same annotation is present, only one invocation is done. Thus, it have exactly one event of type test marking the start and the event making its finish with the correct value.

When the exact same annotation is present multiple times, such the one of this test:
```java
@ValueSource(strings = "foo")
@ValueSource(strings = "foo")
@ValueSource(strings = "foo")
```

Then only one test execution will be triggered instead of three. This is because of how `AnnotationUtils` works in regards to finding repeatable annotations. Investigating the behavior, I've found out the lines 349 & 350 of `findRepeatableAnnotations` method are responsible for unpacking the annotations container.

```java
Annotation[] containedAnnotations = (Annotation[]) ReflectionUtils.invokeMethod(method, candidate);
found.addAll((Collection<? extends A>) asList(containedAnnotations));
```
Given the fact that `found` has a good reason for being a Set (thus removing duplicates), this is why only one test execution is expected.

I find the behavior ok from a user perspective, because it doesn't make sense to annotate a test with multiple `@..Source` in order to achieve the `@RepeatedTest` annotation behavior. 
Let me know if you think the test naming should be adapted or we need to look more into the behavior of test annotated with the exact same annotation (type + value) multiple times.

Reply:
Thanks, I had missed that! I'll discuss it with the team and will get back to you.

Reply:
@marcphilipp, did you got a chance to discuss within the team?

Reply:
Sorry, not yet. We have a call today where I'll bring it up. I'll post here afterwards.